crates/relayer/src/config.rs

```diff
@@ -146,6 +148,62 @@ impl Display for ExtensionOption {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub enum CompatMode {
+    /// Use version 0.34 of the protocol.
+    V0_34,
+    /// Use version 0.37 of the protocol.
+    V0_37,
+}
+
+impl Display for CompatMode {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), FmtError> {
+        match self {
+            Self::V0_34 => write!(f, "v0.34"),
+            Self::V0_37 => write!(f, "v0.37"),
+        }
+    }
+}
+
+impl FromStr for CompatMode {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "V0_34" => Ok(Self::V0_34),
+            "V0_37" => Ok(Self::V0_37),
+            _ => Err(Error::invalid_compat_mode(s.to_owned())),
+        }
```

Comment:
Let's do something like that instead to make it nicer to write in the config:
```suggestion
        const VALID_COMPAT_MODES: &str = "0.34, 0.37";

        // Trim leading 'v', if present
        match s.trim_start_matches('v') {
            "0.34" => Ok(CompatMode::V0_34),
            "0.37" => Ok(CompatMode::V0_37),
            _ => Err(Error::invalid_compat_mode(
                s.to_string(),
                VALID_COMPAT_MODES,
            )),
        }
```

Reply:
done in c31cdae15e2e6028f75d67f11a92371d5f3347e4